ginga/misc/plugins/Cuts.py

```diff
@@ -319,23 +330,54 @@ def stop(self):
 
     def _plotpoints(self, obj, color):
         image = self.fitsimage.get_image()
-        # Get points on the line
-        if obj.kind == 'line':
-            points = image.get_pixels_on_line(int(obj.x1), int(obj.y1),
-                                              int(obj.x2), int(obj.y2))
-        elif obj.kind in ('path', 'freepath'):
-            points = []
-            x1, y1 = obj.points[0]
-            for x2, y2 in obj.points[1:]:
-                pts = image.get_pixels_on_line(int(x1), int(y1),
-                                               int(x2), int(y2))
-                # don't repeat last point when adding next segment
-                points.extend(pts[:-1])
-                x1, y1 = x2, y2
-
-        points = numpy.array(points)
-        self.plot.cuts(points, xtitle="Line Index", ytitle="Pixel Value",
-                       color=color)
+
+        if self.cut_axis_enabled:
+            path = image.get('path', None)
+            hdus = pyfits.open(path, 'readonly')
+            phd = hdus[0].data
```

Comment:
Hmm, I didn't think about that. So should it be something like MultiDim's HDU list? But that would mean opening the file directly which can then be used to access the data.